src/language-js/utils.js

```diff
@@ -945,7 +945,7 @@ function isStringPropSafeToUnquote(node, options) {
       )) ||
       (isSimpleNumber(node.key.value) &&
         String(Number(node.key.value)) === node.key.value &&
-        options.parser === "babel"))
+        (options.parser === "babel" || options.parser === "espree")))
```

Comment:
What about "babel-flow" & "babel-typescript"?

Reply:
https://github.com/prettier/prettier/pull/8508#issue-429331702

Reply:
Got it now. (Comments about Flow & TypeScript seem to be a little scattered. I eventually found the comment about TypeScript in PR #8508.)

I think another nit is to update this comment to include the new `espree` parser:

```js
// So we play it safe and only unquote numbers for the "babel" parser.
```

Reply:
https://github.com/prettier/prettier/pull/9000/commits/a28559b86b66a63bb128caa14641898a51f73fce

Reply:
FYI I have raised the following PRs to test this condition on Angular & Vue parsers:

- #9304
- #9305